packages/wallets/src/features/cfd/components/DynamicLeverageContext/DynamicLeverageContext.tsx

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext } from 'react';
+
+type TDynamicLeverageContext = {
+    isDynamicLeverageVisible: boolean;
+    toggleDynamicLeverage: () => void;
+};
+
+const DynamicLeverageContext = createContext<TDynamicLeverageContext>({
+    isDynamicLeverageVisible: false,
+    toggleDynamicLeverage: () => null,
+});
+
+export const useDynamicLeverageModalState = () => {
+    const context = useContext(DynamicLeverageContext);
```

Comment:
Why are we using context to control state for showing dynamic leverage? 👀

Reply:
Dynamic leverage is not a different modal, we are just flipping the jurisdiction content to show dynamic leverage. To toggle the dynamic leverage from inside the Jurisdiction Card would result in prop drilling, that's why added the context.